Btms.Business/Services/Linking/LinkingService.cs

```diff
@@ -129,11 +131,50 @@ await dbContext.Notifications.Update(notification, notification._Etag, transacti
             }
         }
 
-       
-
         return result;
     }
 
+    private async Task CheckMovementForRemovedLinks(MovementLinkContext linkContext,
+        CancellationToken cancellationToken = default)
+    {            
+        var jp = JsonPath.Parse($"$.{nameof(Movement._MatchReferences)}");
```

Comment:
Just to tidy this up, can we move some of this to ChangeSet as a helper method.

e.g

```
public T? GetPreviousValue<T>(string path)
{
    var jp = JsonPath.Parse($"$.{nameof(Movement._MatchReferences)}");
    var pathResult = jp.Evaluate(Previous);

    if (pathResult.Matches.Any())
    {
        return pathResult.Matches.First().Value.Deserialize<T>();
    }

    return default;
}
```

And then  to get the "Removed" items it can be simplified to

```
var chedRefs = linkContext.ChangeSet?.GetPreviousValue<List<string>>($"$.{nameof(Movement._MatchReferences)}");
var removedRefs = chedRefs .Except(linkContext.PersistedMovement._MatchReferences).ToList()
```